bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingRegistryImpl.java

```diff
@@ -181,4 +186,39 @@ private void notifyTrackerAboutAllThingsRemoved(ThingTracker thingTracker) {
         }
     }
 
+    /**
+     * Removes the {@link Thing} specified by the given {@link ThingUID}.
+     *
+     * If the corresponding {@link ThingHandler} should be given the chance to perform 
+     * any removal operations, use {@link #remove(ThingUID)} instead.
+     * 
+     * @param thingUID Identificator of the {@link Thing} to be removed
+     * @return the {@link Thing} that was removed, or null if no {@link Thing} with the given {@link ThingUID} exists
+     */
+    public Thing forceRemove(ThingUID thingUID) {
+        return super.remove(thingUID);
+    }
+
+    /**
+     * Initiates the removal process for the {@link Thing} specified by the given {@link ThingUID}.
+     * 
+     * Unlike in other {@link Registry}s, {@link Thing}s don't get removed immediately. 
+     * Instead, the corresponding {@link ThingHandler} is given the chance to perform
+     * any required removal handling before it actually gets removed. 
```

Comment:
Maybe we should make more clear, that the method will return before the thing will be removed. If someone wants to listen for the final removal, he can subscribe to the events.